[Docs] AI overview guidelines (#4113)
carbon-design-system/carbon-website: src/pages/guidelines/carbon-for-ai/index.mdx

#### Well-designed explainability:

- Builds trust with users
- Improves understanding of an AI system
- Increases usage and interaction
- Enables product teams to gain further insights to improve the AI system

#### Use the AI label as the path to explainability

The AI label is the primary indicator to communicate that AI is present, and
it’s how users can access explainability content.

#### Start with summarized explanations

As a first layer of explainability, use the explainability pop-over attached to
the AI label to provide a quick, in-context option to explain AI results for a
particular instance.

<Row>
<Column colLg={12}>

<Tabs>

<Tab label="Light mode">

![Light mode UI example of AI explainability shown in a popover.](images/ai-explainability-light.png)

</Tab>

<Tab label="Dark mode">

![Dark mode UI example of AI explainability shown in a popover.](images/ai-explainability-dark.png)

</Tab>

</Tabs>

</Column>
</Row>

### Explainability resources

<Row className="resource-card-group">
<Column colLg={4} colMd={4} noGutterSm>
  <ResourceCard
    subTitle="Designing for Explainable AI"
    href="https://ibm.ent.box.com/v/designingXAI"
    >

> <MdxIcon name="figma" />

Reviewer: ```suggestion
<MdxIcon name="bee" />
```

  </ResourceCard>
</Column>
<Column colLg={4} colMd={4} noGutterSm>
  <ResourceCard
    subTitle="AI Explainability kit (Beta)"
    href="https://www.figma.com/design/jLtMeH3RsHnDs7xF9xUS8f/Carbon-AI---AI-Details-Beta?m=auto&t=Fz8SUpQOgZSNH9Ab-6"
    >

> <MdxIcon name="bee" />

Reviewer: ```suggestion
<MdxIcon name="figma" />
```

I think the icons are switched

Author: oh yeah good catch!

ensure that the AI label remains accessible across light and dark themes, use
the provided component.

#### Light spread is limited for readability

Carbon for AI utilizes light sources to distinguish it from base Carbon
components. These light sources glow on container edges, but the spread is
limited and subtle to ensure accessible contrast with UI content.

<Row>
<Column colLg={8}>

<Tabs>

<Tab label="Light mode">

![Demonstration of text color contrast in a light mode.](images/ai-accessibility-light.png)

</Tab>

<Tab label="Dark mode">

![Demonstration of text color contrast in a dark mode.](images/ai-accessibility-dark.png)

</Tab>

</Tabs>

</Column>
</Row>

## AI resources

<Row className="resource-card-group">

<Column colLg={4} colMd={4} noGutterSm>
  <ResourceCard
    subTitle="IBM Design for AI"
    href="https://w3.ibm.com/w3publisher/design-for-ai"
    actionIcon="launch">
    <MdxIcon name="bee" />
  </ResourceCard>
</Column>
</Row>

## Feedback

Help us improve this component by providing feedback, asking questions, and
leaving any other comments on GitHub. For IBMers, questions and feedback can be

> directed to Slack in the channels #carbon-design-system or #carbon-for-ai.

Reviewer: Should we add links to slack?
[#carbon-design-system](https://ibm-studios.slack.com/messages/C0M053VPT/)
[#carbon-for-ai](https://ibm-studios.slack.com/messages/C0603LZUKRV)

Author: Oh nice those Slack links actually work. I feel like it used to not link very nicely to Slack.

As a first layer of explainability, use the explainability pop-over attached to
the AI label to provide a quick, in-context option to explain AI results for a
particular instance.

<Row>
<Column colLg={12}>

<Tabs>

<Tab label="Light mode">

![Light mode UI example of AI explainability shown in a popover.](images/ai-explainability-light.png)

</Tab>

<Tab label="Dark mode">

![Dark mode UI example of AI explainability shown in a popover.](images/ai-explainability-dark.png)

</Tab>

</Tabs>

</Column>
</Row>

### Explainability resources

<Row className="resource-card-group">
<Column colLg={4} colMd={4} noGutterSm>
  <ResourceCard
    subTitle="Designing for Explainable AI"
    href="https://ibm.ent.box.com/v/designingXAI"
    >

<MdxIcon name="bee" />

  </ResourceCard>
</Column>
<Column colLg={4} colMd={4} noGutterSm>
  <ResourceCard
    subTitle="AI Explainability kit (Beta)"
    href="https://www.figma.com/design/jLtMeH3RsHnDs7xF9xUS8f/Carbon-AI---AI-Details-Beta?m=auto&t=Fz8SUpQOgZSNH9Ab-6"
    >

<MdxIcon name="figma" />

  </ResourceCard>
</Column>
</Row>

## AI label

<InlineNotification>

**Formerly AI slug:** The AI label was previously called "AI Slug" but has been

> updated to better reflect its usage. This name change will be reflected in code

Reviewer 2: Should we also mention that this name change will be reflected in both the code and `Figma` once it moves to stable in the near future?

Author: This is a good point, I'll add a note about the figma component as well.

<ImageCard
sub-title="Date picker"
sub-titleColor="dark"
hoverColor="dark"
href="/components/date-picker/usage/#ai-presence">

![AI date picker](images/mini-date-picker.png)

</ImageCard>

<ImageCard
sub-title="Number input"
sub-titleColor="dark"
hoverColor="dark"
href="/components/number-input/usage/#ai-presence">

![AI number input](images/mini-number-input.png)

</ImageCard>

<ImageCard
sub-title="Text input"
sub-titleColor="dark"
hoverColor="dark"
href="/components/text-input/usage/#ai-presence">

![AI text input](images/mini-text-input.png)

</ImageCard>

</Column>

<Column colMd={3} colLg={3} noGutterSm>

<ImageCard
sub-title="Dropdown"
sub-titleColor="dark"
hoverColor="dark"
href="/components/dropdown/usage/#ai-presence">

![AI dropown](images/mini-dropdown.png)

</ImageCard>

<ImageCard
sub-title="Radio button"
sub-titleColor="dark"
hoverColor="dark"
href="/components/radio-button/usage/#ai-presence">

![AI radio button](images/mini-radio-button.png)

</ImageCard>

<ImageCard
sub-title="Tile"
sub-titleColor="dark"
hoverColor="dark"
href="/components/tile/usage/#ai-presence">

![AI tile](images/mini-tile.png)

</ImageCard>

</Column>

</Row>

## Accessibility

> When designing an AI experience, ask does the user know AI is being used? The

Reviewer 3: ```suggestion
When designing an AI experience, ask if the user knows AI is being used. The
```
